modules/javafx.controls/src/main/java/javafx/scene/control/TextInputControl.java

```diff
@@ -164,8 +164,10 @@ protected TextInputControl(final Content content) {
                 int start = sel.getStart();
                 int end = sel.getEnd();
                 int length = txt.length();
-                if (end > start + length) end = length;
-                if (start > length-1) start = end = 0;
+                // Ensure that the last character to get is within the bounds of the txt string
+                if (end >= start + length) end = length-1;
+                // In case the start is after the whole txt, nothing valid is selected. Thus, return the default.
+                if (start >= length) return "";
```

Comment:
This change seems OK, and might be clearer and the existing code, but the existing code is correct, and produces the same effect.

Reply:
yeah, it is correct - but while we are changing the impl, we might go the whole way and clean up :) my pref would be to add the check for start to the short-circuit at the beginning of the method, something like

        String txt = text.get();
        IndexRange sel = selection.get();
        if (txt == null || sel == null || sel.getStart() >= text.getLength()) return "";